app/src/main/java/com/pvp/app/worker/WeeklyActivityWorker.kt

```diff
@@ -0,0 +1,102 @@
+package com.pvp.app.worker
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.pvp.app.api.ExerciseService
+import com.pvp.app.api.NotificationService
+import com.pvp.app.api.UserService
+import com.pvp.app.model.Notification
+import com.pvp.app.model.NotificationChannel
+import com.pvp.app.model.SportActivity
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.flow.firstOrNull
+
+@HiltWorker
+class WeeklyActivityWorker @AssistedInject constructor(
+    @Assisted context: Context,
+    @Assisted workerParams: WorkerParameters,
+    private val exerciseService: ExerciseService,
+    private val notificationService: NotificationService,
+    private val userService: UserService
+) : CoroutineWorker(
+    context,
+    workerParams
+) {
+
+    companion object {
+
+        const val WORKER_NAME = "WeeklyActivityWorker"
+    }
+
+    override suspend fun doWork(): Result {
+        val activities = getRandomInfrequentActivities()
+
+        postActivityNotification(activities)
+
+        return userService.user
+            .firstOrNull()
+            ?.let { user ->
+                try {
+                    userService.merge(
+                        user.copy(
+                            weeklyActivities = activities
+                        )
+                    )
+
+                    Result.success()
+                } catch (e: Exception) {
+                    Result.failure()
+                }
+            } ?: Result.failure()
+    }
+
+    private fun formNotificationBody(activities: List<SportActivity>): String {
+        return when (activities.count() == 1) {
+            true -> {
+                when (activities.contains(SportActivity.Wheelchair)) {
+                    true -> {
+                        "Participating in Wheelchair activities " +
+                                "this week will give you more points!"
+                    }
+
+                    else -> {
+                        "Participating in ${activities.first().title} " +
+                                "will give you more points this week!"
+                    }
+                }
+            }
+
+            else -> {
+                activities.dropLast(1).joinToString(separator = ", ") { it.title } +
+                        " and " + activities.last().title +
+                        " will give you more points this week! "
+            }
+        }
+    }
+
+    private suspend fun getRandomInfrequentActivities(count: Int = 4): List<SportActivity> {
+        val activities = exerciseService.getInfrequentActivities().toMutableList()
+
+        return when (activities.contains(SportActivity.Wheelchair)) {
```

Comment:
Is this logic fine? Currently, when user has done Wheelchair activities, we drop it and give other activities (that possibly won't exist), else if user hasn't done Wheelchair activities, we give it to the user. Is it right?